src/sagemaker_sklearn_container/mms_patch/mms_transformer.py

```diff
@@ -0,0 +1,175 @@
+2. Not failing when there is no user module, which is needed to support XGBoost algorithm mode
+
+Ideally, these changes, as well as the InferenceToolkitError, should be moved to the SageMaker inference toolkit.
+"""
+
+
+class Transformer(object):
+    """Represents the execution workflow for handling inference requests sent to the model server."""
+    def __init__(self, default_inference_handler=None):
+        """Initialize a ``Transformer``.
+        Args:
+            default_inference_handler (DefaultInferenceHandler): default implementation of inference handlers to
+                use in absence of expected serving functions within the user module.
+                Defaults to ``DefaultInferenceHandler``.
+        """
+        self._default_inference_handler = default_inference_handler or DefaultInferenceHandler()
+        self._initialized = False
+        self._environment = None
+        self._model = None
+
+        self._model_fn = None
+        self._transform_fn = None
+        self._input_fn = None
+        self._predict_fn = None
+        self._output_fn = None
+
+    @staticmethod
+    def handle_error(context, status_code, message):
+        # This model is created to test reporting of an error in a batch of requests
+        context.set_response_status(code=status_code,
+                                    phrase=message,
+                                    idx=0)
+        return [message]
+
+    def transform(self, data, context):
+        """Take a request with input data, deserialize it, make a prediction, and return a
+        serialized response.
+        Args:
+            data (obj): the request data.
+            context (obj): metadata on the incoming request data.
+        Returns:
+            list[obj]: the serialized prediction result wrapped in a list.
+        """
+        if not self._initialized:
+            try:
+                sys_properties = context._system_properties
+                model_dir = sys_properties.get('model_dir')
+
+                self.validate_and_initialize(model_dir)
+            except Exception as e:
+                if isinstance(e, BaseInferenceToolkitError):
+                    logging.error("Error loading model: {}".format(e))
+                    return self.handle_error(context, e.status_code, e.message)
+                else:
+                    raise e
+            self._initialized = True
+
+        try:
+            input_data = data[0].get('body')
+
+            request_processor = context.request_processor[0]
+            request_property = request_processor.get_request_properties()
+            content_type = utils.retrieve_content_type_header(request_property)
+            accept = request_property.get('Accept') or request_property.get('accept')
+
+            if not accept or accept == content_types.ANY:
+                accept = self._environment.default_accept
+
+            if content_type in content_types.UTF8_TYPES:
```

Comment:
If I remember correctly, this was the main difference between the transformer in the sagemaker-inference-toolkit. Have you maybe taken a look to compare this file with the one in that module?

If they have added this in, it might be better to just use their transformer.